src/ExpressionEvaluator/CSharp/Test/ExpressionCompiler/HoistedThisTests.cs

```diff
@@ -950,11 +949,22 @@ static int P
 
         private void VerifyHasThis(string source, string methodName, string expectedType, string expectedIL)
         {
-            var comp = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugDll, assemblyName: ExpressionCompilerUtilities.GenerateUniqueName());
-            var runtime = CreateRuntimeInstance(comp);
+            var sourceCompilation = CreateCompilationWithMscorlib45(source, options: TestOptions.DebugDll, assemblyName: ExpressionCompilerUtilities.GenerateUniqueName());
+            var runtime = CreateRuntimeInstance(sourceCompilation);
             var context = CreateMethodContext(runtime, methodName);
 
             VerifyHasThis(context, expectedType, expectedIL);
+
+            // Now recompile and test CompileExpression with optimized code.
+            sourceCompilation = sourceCompilation.WithOptions(sourceCompilation.Options.WithOptimizationLevel(OptimizationLevel.Release));
```

Comment:
Hmmm...must be getting sloppy (I wasn't running the tests I thought I was).  After latest changes, I realized that I do need a flag (there are cases in the C# tests where "this" isn't optimized away).  Added...